eclass/emboss.eclass

```diff
@@ -37,22 +34,22 @@
 #
 # Defaults to the upstream name of the module.
 
-# @ECLASS-VARIABLE: EBO_EAUTORECONF
-# @DESCRIPTION:
-# Set to 'no', if you don't want eautoreconf to be run after patching.
-: ${EBO_EAUTORECONF:=yes}
-
 # @ECLASS-VARIABLE: EBO_EXTRA_ECONF
 # @DEFAULT_UNSET
 # @DESCRIPTION:
 # Extra config options passed to econf, similar to EXTRA_ECONF.
 
 case ${EAPI:-0} in
-	4) ;;
+	4|5) ;;
 	*) die "this eclass doesn't support < EAPI 4" ;;
 esac
 
-inherit autotools eutils
+if [[ -f "${FILESDIR}"/${P}_fix-build-system.patch ]]; then
+	AUTOTOOLS_AUTORECONF=1
+	AUTOTOOLS_IN_SOURCE_BUILD=1
```

Comment:
Is this only needed when the build_system patch is applied? If so, can't we enhance the patch to support oos builds?